hydrus/data/resource_based_classes.py

```diff
@@ -34,6 +37,39 @@ def get_type(object_: Dict[str, Any]) -> str:
     return type_
 
 
+def get_modified_object(object_: Dict[str, Any], doc: HydraDoc, path: str) -> Dict[str, Any]:
+    """
+    Return the object after modifying properties.
+    :param object_: Dict containing object properties
+    :param doc: HydraDoc object
+    :param path: Path of the Class or Collection
+    :return: object_ containing properties
+    """
+    class_ = doc.parsed_classes[path]
+    properties = class_["class"].supportedProperty
+    datetimefields = []
+    for prop in properties:
+        kwargs = getattr(prop, 'kwargs', None)
+        range = kwargs.get('range')
+        title = getattr(prop, 'title')
+        if range is not None:
+            if "dateTime" in range:
+                datetimefield = title
+                datetimefields.append(datetimefield)
+    if len(datetimefields) != 0:
+        for field in datetimefields:
+            try:
+                datetime_value = object_.get(field)
+                dt_object = parser.isoparse(datetime_value)
+                object_[field] = dt_object
+            except ValueError:
+                raise InvalidDateTimeFormat(field)
+            except TypeError:
+                datetime_value = object_.get(field)
```

Comment:
please add a comment here to specify which error case this exception should handle as it is not clear